### DISCUSSION TITLE
control the first image generated by animatediff

### DISCUSSION THREAD

**Discussion by Li1177** [continue-revolution/sd-webui-animatediff#263] [Q&A]
How can I control the first image generated by animatediff? My current problem is that when I generate a picture by texttoimage , I can get the picture that I'm satisfied with. However, when I choose animatediff to generate a dynamic picture, I find that the first frame of the dynamic picture differs significantly from the picture which I got from texttoimage. Is there a better solution?

**Comment by zappityzap** [comment]
The best method I've found so far is to first generate txt2img without AnimateDiff. Then send the image+prompt to img2img. Enable AnimateDiff and experiment with denoising between 0.7 and 0.9. Lower looks more like the original image, but seems to have less motion. 0.75-0.80 seems best in my limited tests.

ControlNet Canny is better at keeping the look of the original image, but seems to almost eliminate motion. I haven't tested ControlNet settings enough to know if it would work better.
